### PR TITLE
Fix crash in workbench when fitting in script editor

### LINKED ISSUE #27834: Workbench crashes when doing a fit from script editor
### Expected behavior
A fit should be done and mantid should not crash

### Actual behavior
The fit is successful but mantid crash straight after giving the following:
```
Traceback (most recent call last):
  File "c:\users\fvv28776\mantid\qt\python\mantidqt\widgets\codeeditor\completion.py", line 268, in update_completion_api
    self._add_to_completions(self._get_completions_from_globals())
  File "c:\users\fvv28776\mantid\qt\python\mantidqt\widgets\codeeditor\completion.py", line 260, in _get_completions_from_globals
    return generate_call_tips(self.env_globals, prepend_module_name=True)
  File "c:\users\fvv28776\mantid\qt\python\mantidqt\widgets\codeeditor\completion.py", line 203, in generate_call_tips
    if hasattr(f_attr, 'im_func') or inspect.isfunction(f_attr) or inspect.ismethod(f_attr):
  File "c:\users\fvv28776\mantid\framework\pythoninterface\mantid\fitfunctions.py", line 69, in __getattr__
    return self.__getitem__(item)
  File "c:\users\fvv28776\mantid\framework\pythoninterface\mantid\fitfunctions.py", line 103, in __getitem__
    return self.fun.getParameterValue(name)
ValueError: ParamFunction tries to get value of non-existing parameter (im_func) to function Gaussian
Allowed parameters: Height, PeakCentre, Sigma, 
```
 
### Steps to reproduce the behavior
Run the following script:
```
ws = Load("MAR11060")
fitted = Fit("name=Gaussian", ws)
```

### Platforms affected
All

### PR DESCRIPTION
**Pull request by alicerussell1**
**Description of work.**
This fixes the crash when running a script with a fit in it in workbench. This was caused by the move to python 3, specifically a small change in `__hasattr__`, which meant the error was no longer being caught.

**To test:**
Run the following scripts in workbench
```
ws = Load("MAR11060")
fitted = Fit("name=Gaussian", ws)
```

and
```
ws = Load("MAR11060")
fitted = Fit("name=Gaussian;name=LinearBackground", ws)
```
They both should run successfully and not crash.

Fixes #27834 

*This does not require release notes* because **it was caused by the python migration during this sprint**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by PhilColebrooke — CHANGES_REQUESTED** [review]
Mostly good. One small thing, is there any chance you could make it work?

**Review by PhilColebrooke — DISMISSED** [review]
Good to go as long as the checks pass

**Review by PhilColebrooke — DISMISSED** [review]
Still works

File: Framework/PythonInterface/mantid/fitfunctions.py [hunk 1]
```diff
@@ -100,7 +100,10 @@ def __getitem__ (self, name):
         if type(name) == type('string') and self.fun.hasAttribute(name):
             return self.fun.getAttributeValue(name)
         else:
-            return self.fun.getParameterValue(name)
+            if self.fun.hasParameter(name):
+                return self.fun.getParameterValue(name)
+            else:
+                raise AttributeError("Parameter not found")
```
Comment (gvardany) [inline Framework/PythonInterface/mantid/fitfunctions.py]: Could you add the name of the parameter in the error message?

**Comment by gvardany** [comment]
@PhilColebrooke Can you reapprove?

**Review by PhilColebrooke — APPROVED** [review]

**Pull request merged**